xtc/src/ledger.rs

```diff
@@ -140,63 +193,127 @@ impl Ledger {
     }
 }
 
-#[derive(CandidType, Clone, Debug, PartialEq)]
-pub enum APIError {
+//////////////////// BEGIN OF ERC-20 ///////////////////////
+
+#[query(name=balanceOf)]
+pub async fn balance_of(account: Principal) -> Nat {
+    let ledger = ic_kit::get_context().get::<Ledger>();
+    Nat::from(ledger.balance(&account))
+}
+
+#[derive(Deserialize, CandidType)]
+pub struct TransferArguments {
+    pub to: Principal,
+    pub amount: u64,
+}
+
+#[derive(CandidType, Debug)]
+pub enum TransferError {
     InsufficientBalance,
-    InsufficientAllowance,
+    AmountTooLarge,
+    CallFailed,
     Unknown,
 }
 
-#[derive(CandidType, Clone, Debug)]
-pub struct ErrorDetails {
-    msg: &'static str,
-    code: APIError,
+#[query]
+pub async fn allowance(from: Principal, to: Principal) -> Nat {
+    return get_context().get::<Ledger>().allowance(&from, &to).into();
 }
 
-static InsufficientBalanceError: ErrorDetails = ErrorDetails {
-    msg: "Insufficient Balance",
-    code: APIError::InsufficientBalance,
-};
+#[update]
+pub async fn approve(to: Principal, amount: Nat) -> TxReceipt {
+    IsShutDown::guard();
+    use ic_cdk::export::candid;
+    let ic = get_context();
```

Comment:
warning: use of deprecated function `ic_kit::get_context`: get_context is deprecated use ic_kit::ic::*

I can't make a GitHub suggestion due to how it works, otherwise I would.